src/vos/vos_aggregate.c

```diff
@@ -1524,6 +1524,10 @@ insert_segments(daos_handle_t ih, struct agg_merge_window *mw, bool last, unsign
 
 	/** Remove processed removal records */
 	rc = process_removals(mw, oiter, &mw->mw_rmv_ents, last, true);
+	if (rc != 0) {
+		DL_ERROR(rc, "Unable to process extent removals");
+		goto abort;
+	}
```

Comment:
Could you explain a bit why it this rebuild overwrite change could affect removal records?

Reply:
It can't but this change can avoid an assertion failure.

Reply:
Should we add a new assert in process_removals() (like asserting evt_delete() won't return -DER_ENOENT) or somewhere else?